modules/dasBidAdapter.js

```diff
@@ -0,0 +1,250 @@
+      customParams.DV = pageContext.dv;
+    }
+
+    // Keywords
+    const keywords = getAllOrtbKeywords(
+      bidderRequest?.ortb2,
+      pageContext.keyWords,
+    );
+    if (keywords.length > 0) {
+      customParams.kwrd = keywords.join('+');
+    }
+
+    // Local capping
+    if (pageContext.capping) {
+      customParams.local_capping = pageContext.capping;
+    }
+
+    // Key values
+    if (pageContext.keyValues && typeof pageContext.keyValues === 'object') {
+      Object.entries(pageContext.keyValues).forEach(([key, value]) => {
+        keyValues[`kv${key}`] = value;
+      });
+    }
+  }
+
+  const du = customParams.du || deepAccess(bidderRequest, 'refererInfo.page');
+  const dr = customParams.dr || deepAccess(bidderRequest, 'refererInfo.ref');
+
+  if (du) customParams.du = du;
+  if (dr) customParams.dr = dr;
+
+  const dsaRequired = deepAccess(bidderRequest, 'ortb2.regs.ext.dsa.required');
+  if (dsaRequired !== undefined) {
+    customParams.dsainfo = dsaRequired;
+  }
+
+  return {
+    customParams,
+    keyValues,
+  };
+}
+
+function buildUserIds(customParams) {
+  const userIds = {};
+  if (customParams.lu) {
+    userIds.lu = customParams.lu;
+  }
+  if (customParams.aid) {
+    userIds.aid = customParams.aid;
+  }
+  return userIds;
+}
+
+function getNpaFromPubConsent(pubConsent) {
+  const params = new URLSearchParams(pubConsent);
+  return params.get('npa') == '1';
+}
+
+function buildOpenRTBRequest(bidRequests, bidderRequest) {
+  const { customParams, keyValues } = parseParams(
+    bidRequests[0].params,
+    bidderRequest,
+  );
+  const imp = bidRequests.map((bid, index) => {
+    const sizes = getAdUnitSizes(bid);
+    const imp = {
+      id: bid.bidId,
+      tagid: bid.params.slot,
+      secure: 1,
+      ext: {
+        pos: bid.params.slotSequence || 0,
+      },
+    };
+
+    if (bid.mediaTypes?.banner) {
+      imp.banner = {
+        format: sizes.map((size) => ({
+          w: size[0],
+          h: size[1],
+        })),
+      };
+    }
+
+    return imp;
+  });
+
+  const request = {
+    id: bidderRequest.bidderRequestId,
+    imp,
+    site: {
+      id: customParams.site,
+      page: customParams.du,
+      ref: customParams.dr,
+      ext: {
+        area: customParams.area,
+        kwrd: customParams.kwrd,
+        dv: customParams.DV
+      },
+      ...bidderRequest.ortb2.site,
+    },
+    device: bidderRequest.device,
+    user: {
+      ext: {
+        ids: buildUserIds(customParams),
+      },
+    },
+    ext: {
+      network: customParams.network,
+      keyvalues: keyValues,
+    },
+    at: 1,
+    tmax: bidderRequest.timeout
+  };
+
+  if (bidderRequest.gdprConsent) {
+    request.user = {
+      ext: {
+        dsa: customParams.dsainfo,
+        npa: getNpaFromPubConsent(customParams.pubconsent),
+        localcapping: customParams.local_capping,
+        ...request.user.ext,
+      },
+    };
+    request.regs = {
+      gdpr: bidderRequest.gdprConsent.gdprApplies,
```

Comment:
Czy ta wartość na pewno jest 0/1 (jak wymaga specyfikacja OpenRTB), a nie true/false? Może na wszelki wypadek zrobić tak:
```suggestion
      gdpr: bidderRequest.gdprConsent.gdprApplies ? 1 : 0,
```